### ISSUE TITLE
Change 'choose local (ours)' to 'choose local (branch-name)' request

### ISSUE THREAD

**Issue by yaustar** [gitextensions/gitextensions#6582]
## Feature description
We use Git in games where there are some non technical people working on the team. 

To make merge conflicts clearer, could there be an option to replace the ours/theirs in the brackets be the branch name instead? eg.

Currently:
Choose local (ours)
Choose remote (theirs)

To:
Choose local (ours/master)
Choose remote (theirs/feature-branch)

This is mostly to give confidence to our non-technical users of which change they want as typically they are dealing with binary assets.

Thanks in advance.

**Comment by gerhardol** [comment]
Reasonable to display the branches here.

I would prefer keeping ours/theirs separated, that is the Git names, local/remote are the GE "decorations".
Choose origin/master (ours) 
Choose my/feature-branch (theirs)

Only seem to affect FormResolveConflicts.cs

**Comment by pmiossec** [comment]
I also like the idea to display the branch name.

**Comment by RussKie** [comment]
I like the idea in general 👍 I wholeheartedly agree the concept can be quite confusing even to those who use git on a regular basis - depending on an operation "ours" can mean a local or a remote branch, and one has to think what it means every time.

However whatever name/labels we choose we need to be consistent with a) git terminology and b) with all diff tools users may be using.

**Comment by yaustar** [comment]
Could it be an option to show the branch name (settings/part of the merge dialog)? 

There's the case where people are working on the same branch and have to merge which can lead to some confusion. eg:

Choose local (ours/master)
Choose remote (theirs/master)

Which can seem odd the first time people seeing it.

**Comment by pmiossec** [comment]
>There's the case where people are working on the same branch and have to merge which can lead to some confusion.

I don't think this reason is enough to introduce a setting that add code complexity and that no one will change.

And that we could be clever enough to either display nothing, in this special case where branches names are the same, or find a good wording to say "commits rebased".

**Comment by RussKie** [comment]
I agree, we need to provide more information. Granted, you may not have a branch name (local or remote) available for _local_, _base_ or _remote_ revision. But we must at least show a hash for each.

    Local   src/Common/src/Some/Directory/file.cs   (ours/master, as098df0)
    Base    src/Common/src/Some/Another/code.cs     (2v35mj23)
    Remote  deleted                                 (ours/branch, 23lmkl23)

I'm currently staring at the following and have no idea which one is which...
![image](https://user-images.githubusercontent.com/4403806/60151412-fa2a6000-981f-11e9-8a19-c7c3285ed049.png)

**Comment by RussKie** [comment]
Somewhat related: #6863, should be an easy fix

**Comment by RussKie** [comment]
SemanticMerge shows a handy visual:
![image](https://user-images.githubusercontent.com/4403806/66019201-a2d61300-e525-11e9-8831-2260470d8d54.png)

**Comment by RussKie** [comment]
Closing as stale.
There has been some adjustments made in [#8349](https://github.com/gitextensions/gitextensions/issues/8349)
